docs/developer-guide/ksqldb-clients/contributing.md

```diff
@@ -0,0 +1,175 @@
+| List Kafka topics           | LIST TOPICS;       |
+| List running ksqlDB queries | LIST QUERIES;      |
+
+The [server endpoint documentation](../ksqldb-rest-api/ksql-endpoint.md)
+contains the format of the server response body for each type of request. For example, the response of
+a `LIST STREAMS;` command is a JSON object with an array-type field with name `streams`. Each element
+of the array is a JSON object with fields `name`, `topic` , `format`, and `type`, each of type string.
+
+### Creation and deletion of streams and tables ###
+
+These DDL/DML statements include:
+- Creation of source streams and tables: `CREATE STREAM`, `CREATE TABLE`
+- Creation of persistent queries: `CREATE STREAM AS SELECT`, `CREATE TABLE AS SELECT`
+- Deletion of streams and tables: `DROP STREAM`, `DROP TABLE`
+
+Send these DDL/DML requests to the [`/ksql` server endpoint](../ksqldb-rest-api/ksql-endpoint.md).
+
+As explained in the [server endpoint documentation](../ksqldb-rest-api/ksql-endpoint.md),
+the HTTP request body contains the text of the ksqlDB statement as well as any relevant
+query properties. Because ksqlDB supports many different types of statements for creating streams,
+tables, and persistent queries, the client interface should accept a SQL statement and pass the statement
+to the server endpoint directly.
+
+The server response will contain a command ID, command sequence number, and status information.
+These pieces of information are typically do not inform actions of ksqlDB users and do not need to be
+exposed as part of the client interface as a result.
+
+Note that the `/ksql` server endpoint accepts not only DDL/DML requests as described here, but also
+other types of statements including admin operation requests described [above](#listing-existing-streams-tables-topics-and-queries).
+If the server response for a SQL statement submitted to the `/ksql` endpoint has HTTP status code 200
+but the response body does not fit the form documented above, then the submitted SQL statement did
+not correspond to a DDL/DML statement. The client should handle this potential scenario by returning
+an error to the user.
+
+### Terminating persistent queries ###
+
+Send these requests to the [`/ksql` server endpoint](../ksqldb-rest-api/ksql-endpoint.md).
+
+The SQL statement that should be sent in the HTTP request body is `TERMINATE <persistent_query_id>;`.
+
+As noted in the [server endpoint documentation](../ksqldb-rest-api/ksql-endpoint.md),
+the format of the response body for terminate query requests is the same as for other DDL/DML requests
+[above](#creation-and-deletion-of-streams-and-tables).
+
+Configurations
+--------------
+
+The client must expose options for specifying the address of the ksqlDB server to connect to.
+Additional options that are important to expose include:
+- Support for TLS-enabled ksqlDB servers
+- Support for mutual-TLS-enabled ksqlDB servers
+- Support for ksqlDB servers with HTTP basic authentication enabled
```

Comment:
support for ksqlDB servers with custom auth?

Reply:
What kind of custom auth are you thinking of? If we're talking about situations where clients need to pass custom HTTP auth headers, those aren't currently supported by the Java client. The way to support them would be to add an option to specify custom headers to be added to all requests.

Reply:
The server supports a plugging in custom authentication mechanisms.  This means someone could write a client that was compatible. Not sure what that means in the context of this document. Maybe its sufficient to just call out this is possible.

Reply:
Ack. Clarified along with the wording change suggested below (regarding supporting multiple ksqlDB server addresses).